diff --git a/src/Services/LocalFilesManager.php b/src/Services/LocalFilesManager.php
index cf8c0ae..e0a01ad 100644
--- a/src/Services/LocalFilesManager.php
+++ b/src/Services/LocalFilesManager.php
@@ -228,7 +228,7 @@ class LocalFilesManager {
       'theme' => "theme/{$themeFileName}-stylesheet.min.css",
       'grid' => "cohesion-responsive-grid-settings.css",
       'icons' => "cohesion-icon-libraries.css",
-      'prefixed' => "prefixed/cohesion-prefixed-ckeditor-stylesheet.css",
+      // 'prefixed' => "prefixed/cohesion-prefixed-ckeditor-stylesheet.css",
       'preview' => "preview/cohesion-custom-style-stylesheet.css",
     ];
 
@@ -238,7 +238,7 @@ class LocalFilesManager {
       'theme' => "{$themeFileName}-theme-stylesheet.min.css",
       'grid' => "cohesion-responsive-grid-settings.css",
       'icons' => "cohesion-icon-libraries.css",
-      'prefixed' => "cohesion-prefixed-ckeditor-stylesheet.css",
+      // 'prefixed' => "cohesion-prefixed-ckeditor-stylesheet.css",
       'preview' => "cohesion-custom-style-stylesheet.css",
     ];
 
